### PR TITLE
2015 enrich settings with background processing

### PR DESCRIPTION
**Pull request by marcussc**
## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description
Solves: https://jira.itc.sap.com/browse/EXPOSUREAPP-2015

Simulator Screen Shot - iPhone 11 Pro Max - 2020-08-20 at 12 18 30
Simulator Screen Shot - iPhone 11 Pro Max - 2020-08-20 at 12 18 32
Simulator Screen Shot - iPhone 11 Pro Max - 2020-08-20 at 12 18 56
Simulator Screen Shot - iPhone 11 Pro Max - 2020-08-20 at 12 19 06
Screen Shot 2020-08-20 at 12 20 33

Draft: Waiting for a11y text for the image.

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+					self?.present(activityViewController, animated: true, completion: nil)
+				}
+			}
+		)
+	}
+	
+	// MARK: - Overrides
+	
+	override func viewDidLoad() {
+		setupView()
```
Comment (aleksandr-tikhonov) [inline src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift:49]: you should call super here

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			viewModel.$infoBoxViewModel.receive(on: RunLoop.main).sink { [weak self] in
+				self?.updateInfoxBox(with: $0)
+			}
+		]
+		
+	}
+	
+	private func updateInfoxBox(with viewModel: InfoBoxViewModel?) {
+		if let viewModel = viewModel {
+			if !contentStackView.arrangedSubviews.contains(infoBox) { contentStackView.addArrangedSubview(infoBox) }
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift:102]:
```suggestion
			if infoBox.superview == nil { contentStackView.addArrangedSubview(infoBox) }
```

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/InfoBoxView.swift [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		])
+
+		let enumerationLabel = ENALabel()
+		enumerationLabel.text = "\(index + 1)."
+		enumerationLabel.numberOfLines = 1
+		enumerationLabel.style = .headline
+		enumerationLabel.setContentCompressionResistancePriority(.required, for: .horizontal)
+
+		containerView.addSubview(enumerationLabel)
+		enumerationLabel.translatesAutoresizingMaskIntoConstraints = false
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/InfoBoxView.swift:130]: Doesn't it make more sense to set `translatesAutoresizingMaskIntoConstraints` to `false` before adding it to `containerView`?

Comment (ChristianKienle) [reply src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/InfoBoxView.swift:130]: Same for the other views that are added prior to setting `translatesAutoresizingMaskIntoConstraints` to `false`.

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		didSet {
+			calculateCombinedBackgroundStatus()
+		}
+	}
+
+	private var combinedBackgroundAppRefreshStatus: CombinedBackgroundAppRefreshStatus = .off {
+		didSet {
+			switch combinedBackgroundAppRefreshStatus {
+			case .on:
+				backgroundAppRefreshStatusText = AppStrings.BackgroundAppRefreshSettings.Status.on
```
Comment (aleksandr-tikhonov) [inline src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewModel.swift:111]: Do we need here `showInfoBox = false`?

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+			},
+			viewModel.$backgroundAppRefreshStatusAccessibilityLabel.receive(on: RunLoop.main).sink { [weak self] in
+				self?.backgroundAppRefreshStatusStackView.accessibilityLabel = $0
+			},
+			viewModel.$image.receive(on: RunLoop.main).sink { [weak self] in
+					self?.imageView.image = $0
+			},
+			viewModel.$infoBoxViewModel.receive(on: RunLoop.main).sink { [weak self] in
+				self?.updateInfoxBox(with: $0)
+			}
```
Comment (aleksandr-tikhonov) [inline src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift:95]: I like that finally we have some `Combine` code in the project 👍

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift (hunk 1 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewController.swift:50]: Added :)

**Comment by aleksandr-tikhonov** [comment]
Also please check the dynamic type on the screens, that the app with small and large fonts looks fine. (`ENALabel` should support it)

**Comment by marcussc** [comment]
> Also please check the dynamic type on the screens, that the app with small and large fonts looks fine. (`ENALabel` should support it)

Done, Voice Over has been tested as well :)

**Comment by haosap** [comment]
Hi @marcussc ,
Very nice 1st PR 🥇 
Have you also considered the situation that the user jump out of the app to turn on/off the low power mode? Thanks. I see you don't use the `powerStateChanged`.

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewModel.swift (hunk 4 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/BackgroundAppRefreshViewModel.swift:111]: Nice finding, this was an old property which is no longer used. 
I removed `@Published var showInfoBox: Bool = false`

**Comment by marcussc** [comment]
> Hi @marcussc ,
> Very nice 1st PR 🥇
> Have you also considered the situation that the user jump out of the app to turn on/off the low power mode? Thanks. I see you don't use the `powerStateChanged`.

Yes this has been considered:
https://github.com/corona-warn-app/cwa-app-ios/pull/1068/files#diff-41365a72919dd28f14dd3faf60a88edeR190

**Comment by haosap** [comment]
> > Hi @marcussc ,
> > Very nice 1st PR 🥇
> > Have you also considered the situation that the user jump out of the app to turn on/off the low power mode? Thanks. I see you don't use the `powerStateChanged`.
> 
> @haosap Yes this has been considered:
> https://github.com/corona-warn-app/cwa-app-ios/pull/1068/files#diff-41365a72919dd28f14dd3faf60a88edeR190

Thanks. I oversaw it.

File: src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/InfoBoxView.swift (hunk 3 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/Settings/BackgroundAppRefresh/InfoBoxView.swift:129]: Changed.

TBH: Didn't knew that the order makes a difference :)

**Review by aleksandr-tikhonov — APPROVED** [review]

**Review by janetback — APPROVED** [review]
Texts reviewed by UA.

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**